src/terminal/TerminalClientMain.cpp

```diff
@@ -241,7 +245,9 @@ int main(int argc, char** argv) {
     TerminalClient terminalClient(
         clientSocket, clientPipeSocket, socketEndpoint, id, passkey, console,
         is_jumphost, result.count("t") ? result["t"].as<string>() : "",
-        result.count("r") ? result["r"].as<string>() : "", result.count("f"));
+        result.count("r") ? result["r"].as<string>() : "",
+        (result.count("f") || sshConfigOptions.forward_agent),
+        result.count("ssh-socket") ? result["ssh-socket"].as<string>() : string(sshConfigOptions.identity_agent));
```

Comment:
This looks suspicious, could this end up being string(NULL)?

Reply:
Yes, yes indeed it could, thank you. I'll rebuild a fix for that and update the commit.